addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java

```diff
@@ -153,20 +156,32 @@ public ZestResponse runStatement(
                     }
                 }
 
-                steps.add(
+                SequenceStepData stepData =
                         new SequenceStepData(
                                 step,
                                 passed,
                                 result,
                                 ascan.getAlertsIds(),
                                 ZestZapUtils.toHttpMessage(req, req.getResponse()),
-                                msg));
+                                msg);
+                steps.add(stepData);
+                countStepStats(stepData);
             }
         }
 
         return resp;
     }
 
+    private static void countStepStats(SequenceStepData step) {
+        String ascanStep = ExtensionSequenceAutomation.STATS_PREFIX + "ascan.step" + step.getStep();
```

Comment:
Looks good but we should now use a more generic constant for these outside the automation.